x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -407,6 +420,52 @@ private LogicalPlan resolveAggregate(Aggregate a, List<Attribute> childrenOutput
             return a;
         }
 
+        private LogicalPlan resolveLookup(EsqlConfiguration config, Lookup p, List<Attribute> childrenOutput) {
+            if (p.expressionsResolved()) {
+                return p;
+            }
+            String tableName = p.tableName().name();
+            Map<String, Column> table = config.tables().get(tableName);
+            if (table == null) {
+                return p;
+            }
+            table = new TreeMap<>(table);
+            // TODO what's the right type for this thing?!
+            Attribute resolvedTableName = new MetadataAttribute(p.tableName().source(), tableName, KEYWORD, false);
```

Comment:
I have no idea what the appropriate `Attribute` subclass is here - we don't really use it for anything after this other than, like, `toString` and making sure all of the attributes are `resolved`.